### ISSUE TITLE
Add support for WASI 0.2

### ISSUE THREAD

**Issue by yoshuawuyts** [smol-rs/fastrand#89]
The WASI 0.2 target will reach tier 2 in Rust 1.82. This crate supports WASI 0.1, and it would be neat if it could support WASI 0.2 as well.

I looked into it briefly, and I believe this depends on https://github.com/rust-random/getrandom/pull/499 landing first since that's used internally. Landing this in turn should enable us to update the `polling` crate, which uses `fastrand` in its test suite.

**Comment by taiki-e** [comment]
Since it has a valid `std::time::Instant`, this crate does nothing special with WASI.

**Comment by yoshuawuyts** [comment]
Yeah, that seems about right. When I first filed this issue support for that wasn't in the stdlib, but happy to close it now!
